### PR TITLE
Send heartbeat after one minute if "receivedAnyStats: false"

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/background_process/heartbeats.py [hunk 1]
```diff
@@ -32,5 +32,8 @@ def send_heartbeat_wrapper(rep, interval_in_secs, event_scheduler):
         send_heartbeat_wrapper,
         (rep, interval_in_secs, event_scheduler),
     )
-    logger.debug("Heartbeat...")
-    rep.send_heartbeat()
+    if not boot:
+        #  If boot is true it means it's the first time this gets executed, so we
+        #  need to wait for the interval in the event scheduler to finish
+        logger.debug("Heartbeat...")
```
Comment (willem-delbare) [inline aikido_firewall/background_process/heartbeats.py:38]: odd varname. If boot is false, then it does boot the heartbeats?

File: aikido_firewall/background_process/reporter/__init__.py [hunk 2]
```diff
@@ -55,10 +55,20 @@ def start(self, event_scheduler):
                 "Token was invalid, not starting heartbeats and realtime polling."
             )
             return
+        event_scheduler.enter(60, 1, self.report_initial_stats)
```
Comment (willem-delbare) [inline aikido_firewall/background_process/reporter/__init__.py]: this is the config-last-update-time-poll-interval? let's move it to a var at the top

File: aikido_firewall/vulnerabilities/__init__.py [hunk 3]
```diff
@@ -74,11 +74,13 @@ def run_vulnerability_scan(kind, op, args):
...
     elif kind == "ssrf":
         # Report hostname and port to background process :
-        get_comms().send_data_to_bg_process("HOSTNAMES_ADD", (args[0], args[1]))
         injection_results = check_context_for_ssrf(
             hostname=args[0], port=args[1], operation=op, context=context
         )
         error_type = AikidoSSRF
+        blocked_request = is_blocking_enabled() and injection_results
+        if not blocked_request:
+            get_comms().send_data_to_bg_process("HOSTNAMES_ADD", (args[0], args[1]))
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/__init__.py]: kind of odd this code has to know about blocking not being enabled

Comment (bitterpanda63) [reply aikido_firewall/vulnerabilities/__init__.py]: unrelated, merged main into the other branch

**Pull request closed without merging**